### PR TITLE
Interactive Entities

### PR DESCRIPTION
**Pull request by austin0209**
Add interactive entities. We still need to add spikes to allow for complete level creation.

### CODE REVIEW

File: src/ecs/components.h [hunk 1]
```diff
@@ -31,6 +32,7 @@
 #define LAYER_NONE ((u64)0)
 #define LAYER_TERRAIN ((u64)1 << 0)
 #define LAYER_LETHAL ((u64)1 << 1)
+#define LAYER_INTERACTABLE ((u64)1 << 1)
```
Comment (austin0209) [inline src/ecs/components.h]: Do we need this new layer? Not sure how I should be thinking about the layers, but I don't see the battery or solar panels fitting in any of the other layers.

File: src/ecs/entities/battery.c [hunk 2]
```diff
@@ -0,0 +1,72 @@
+#include "../../atlas.h"
+#include "battery.h"
+#include "common.h"
+#include <raymath.h>
+
+EntityBuilder BatteryCreate(const f32 x, const f32 y, const AtlasSprite* atlasSprite)
```
Comment (austin0209) [inline src/ecs/entities/battery.c:6]: Is this how you intended the Atlas API to be used (passing in atlasSprite)?

File: src/ecs/components.h (hunk 1 above)
Comment (thismarvin) [reply src/ecs/components.h]: We definitely need a new layer.

I'll get back to you on how we should think of layers moving forward.

**Review by thismarvin — APPROVED** [review]
Interfacing with `AtlasSprite`/`CSprite` will change soon, but for now everything looks good!

**Pull request merged**